lib/init/grass.py

```diff
@@ -1915,52 +1926,77 @@ def bash_startup(location, location_name, grass_env_file):
     home = location                   # save .bashrc in $LOCATION
     os.environ['HOME'] = home
 
-    bashrc = os.path.join(home, ".bashrc")
-    try_remove(bashrc)
+    shell_rc_file = os.path.join(home, shrc)
+    try_remove(shell_rc_file)
 
-    f = open(bashrc, 'w')
-    f.write("test -r ~/.alias && . ~/.alias\n")
+    f = open(shell_rc_file, 'w')
+    
+    if shell_name == 'zsh':
+        f.write('test -r {home}/.alias && source {home}/.alias\n'.format(
+            home=userhome))
```

Comment:
Does it mean `zsh` doesn't support `test -r ~/.alias && . ~/.alias`? Interesting.

Reply:
As I understand this, this line is written to a `.*rc` file in the new $HOME at $LOCATION. `~` is thus expanded to (the new) $HOME where there is no .alias file. This is the case, at least for mac, with both bash and zsh.

Reply:
To be concise:
`test -r {home}/.alias && . {home}/.alias` does also work. `source` not necessary (a period `.` works too), but expanded (original user) home is needed.

Reply:
> As I understand this, this line is written to a `.*rc` file in the new $HOME at $LOCATION. `~` is thus expanded to (the new) $HOME where there is no .alias file. This is the case, at least for mac, with both bash and zsh.

I see. You decided to source `/home/USER/.alias` from the startup script. For bash, it's up to the user and that's why I didn't do `source /home/USER/.alias` here. The user can always include `source ~/.alias` in their `~/.grass7/bashrc` if they want. I gave them a choice between a GRASS-specific bash environment vs. a full regular bash environment. That's a design decision.

Reply:
> I see. You decided to source /home/USER/.alias from the startup script. For bash, it's up to the user and that's why I didn't do source /home/USER/.alias here. The user can always include source ~/.alias in their ~/.grass7/bashrc if they want. I gave them a choice between a GRASS-specific bash environment vs. a full regular bash environment. That's a design decision.

I understand .alias can be sourced in ~/.grass7/bashrc, with which I have no fundamental problem with. But as of now that line https://github.com/OSGeo/grass/blob/c3a3f57aaff7acdc08fcbb82ed6647c91f0b7ce2/lib/init/grass.py#L1922 doesn't do anything and may be removed completely or alternatively rewrite it as I did now for zsh.

Reply:
> > I see. You decided to source /home/USER/.alias from the startup script. For bash, it's up to the user and that's why I didn't do source /home/USER/.alias here. The user can always include source ~/.alias in their ~/.grass7/bashrc if they want. I gave them a choice between a GRASS-specific bash environment vs. a full regular bash environment. That's a design decision.
> 
> I understand .alias can be sourced in ~/.grass7/bashrc, with which I have no fundamental problem with. But as of now that line
> 
> https://github.com/OSGeo/grass/blob/c3a3f57aaff7acdc08fcbb82ed6647c91f0b7ce2/lib/init/grass.py#L1922
> doesn't do anything and may be removed completely or alternatively rewrite it as I did now for zsh.

I don't know about the history of that line, but creating `$LOCATION/.alias` works fine.

Reply:
> I don't know about the history of that line, but creating $LOCATION/.alias works fine.

True, if there is a file there it will work, but by default there is none and it has to be filled with content. I'm wondering, is it a common practice for (some) pro-users to create specific .alias file for a single location?
Sourcing from /Users/theuser/.grass7/bashrc makes sense to me, but creating location specific .alias files not so much. But I may of course overlook something.

Reply:
I understand. Probably, it was an oversight by the original author and he intended to source `/home/USER/.alias`. Or it might have been that way until the `$HOME` hack broke it? Who knows... ;-)

Reply:
I could trace that line to at least [GRASS 5.0](https://github.com/OSGeo/grass-legacy/blob/e3cbac3817101039d6d98a69d3a09dd944c4fce4/src/general/init/init.sh#L564).

As I understand that code the $HOME hack was in place already, I'm curious whether the .alias line worked back then.

Reply:
Bash stays as is, while I choose to source alias for zsh.